sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOIT.java

```diff
@@ -196,11 +204,18 @@ public void testPublishingThenReadingAll() throws IOException {
     MetricsReader metricsReader = new MetricsReader(readResult, NAMESPACE);
     long actualRecords = metricsReader.getCounterMetric(READ_ELEMENT_METRIC_NAME);
 
+    // TODO(yathu) resolve pending messages with direct runner. Due to direct runner only finalize
```

Comment:
Feel free to dismiss if this is not feasible; I don't understand enough about JMS to know whether what I'm asking here is. Was wondering if this PR could add:
1) failing test that attempts to resolve pending messages
2) Add `@Ignore` annotation to test and reference a GitHub issue that aims to resolve failing test.